packages/opencensus-instrumentation-http/test/test-http.ts

```diff
@@ -102,40 +107,143 @@ describe('HttpPlugin', function() {
+             assert.strictEqual(
+                 span.attributes[plugin.ATRIBUTE_HTTP_METHOD], 'GET');
+             assert.strictEqual(
+                 span.attributes[plugin.ATRIBUTE_HTTP_PATH], testPath);
+             assert.strictEqual(
+                 span.attributes[plugin.ATRIBUTE_HTTP_ROUTE], testPath);
+             assert.strictEqual(
+                 span.attributes[plugin.ATRIBUTE_HTTP_USER_AGENT], null);
+             assert.strictEqual(
+                 span.attributes[plugin.ATRIBUTE_HTTP_STATUS_CODE],
+                 httpErrorCodes[i].toString());
+           });
+         });
+    }
+
+
     it('should create a child span for GET requests', () => {
       const testPath = '/outgoing/rootSpan/childs/1';
-      doNoke(urlHost, testPath, 200, 'Ok');
+      doNock(urlHost, testPath, 200, 'Ok');
       const options = {name: 'TestRootSpan'};
       return tracer.startRootSpan(options, async root => {
         await httpRequest.get(`${urlHost}${testPath}`).then((result) => {
           assert.ok(root.name.indexOf('TestRootSpan') >= 0);
           assert.strictEqual(root.spans.length, 1);
           assert.ok(root.spans[0].name.indexOf(testPath) >= 0);
           assert.strictEqual(root.traceId, root.spans[0].traceId);
+          const span = root.spans[0];
+          assert.strictEqual(span.status, plugin.traceStatus(200));
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_HOST], hostName);
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_METHOD], 'GET');
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_PATH], testPath);
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_ROUTE], testPath);
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_USER_AGENT], null);
+          assert.strictEqual(
+              span.attributes[plugin.ATRIBUTE_HTTP_STATUS_CODE], '200');
         });
       });
     });
 
-    // TODO: testing failing for node 6
+    for (let i = 0; i < httpErrorCodes.length; i++) {
+      it(`should test a child spans for GET requests with http error ${
+             httpErrorCodes[i]}`,
+         () => {
+           const testPath = '/outgoing/rootSpan/childs/1';
+           doNock(
+               urlHost, testPath, httpErrorCodes[i],
+               httpErrorCodes[i].toString());
+           const options = {name: 'TestRootSpan'};
+           return tracer.startRootSpan(options, async root => {
+             await httpRequest.get(`${urlHost}${testPath}`).then((result) => {
+               assert.ok(root.name.indexOf('TestRootSpan') >= 0);
+               assert.strictEqual(root.spans.length, 1);
+               assert.ok(root.spans[0].name.indexOf(testPath) >= 0);
+               assert.strictEqual(root.traceId, root.spans[0].traceId);
+
+               const span = root.spans[0];
+               assert.strictEqual(
```

Comment:
Suggestion: A lot of this code looks the same. For maintainability it would be a good idea to see what can be factored out into a common function.

Reply:
Done!